Report/Template.Rmd

```diff
@@ -243,17 +251,262 @@ datasets = read.and.clean(dataset = "easySHARE_rel6_0_0.rda", wav = 99)
+            val = sum(x)/length(x)
+            return(val)
+        })
+    } else { # return total if country-level aggregation not specified
+        final = sum(df[[y]])/length(df[[y]])
+    }
+    return(final)
+}
+
+# Function for calculating labor participation rate per country / gender / age
+labor = function(df, .country, .gender, .age) {
+    dataset = df %>% 
+        filter(country == .country & gender == .gender & get(.age) == TRUE)
+    denom = nrow(dataset)
+    numer = with(dataset, sum(labor_ft | labor_pt))
+    perc  = round(numer/denom, 4)
+    
+    return(perc)
+}
+
+```
+
+Within the code below, `args.list` contains three sub-lists divided by countriy (`c`), gender (`g`) and age (`a`). The list is passed as an argument to `labor`, which are simulatenously iterated over the sub-lists using `pmap`. This function has an advantage over a similar functions, since it allows multiple arguments and uses parallelization by default.  
+Including `with` to specify the data frame on which we expand the grid upon, modifies the data frame in a new environment. This simplifies name assignment. 
+
+```{r, eval = FALSE}
+
+# use expand.grid to get all combinations of country / gender / age
+agegroups = names(df.out[, 4:6])
+args      = with(df.out, 
+                 expand.grid(levels(country), levels(gender), agegroups, 
+                             stringsAsFactors = FALSE))
+args.list = list(c = as.vector(t(args[1])), 
+                 g = as.vector(t(args[2])), 
+                 a = as.vector(t(args[3])))
+
+# Map over multiple arguments
+output   = pmap(args.list, function(c, g, a) labor(df.out, c, g, a))
+output.v = do.call("c", output)
+
+labor.part.share    = cbind(args, output.v)
+labor.part.share.df = spread(labor.part.share, Var3, output.v) %>% 
+    arrange(desc(Var2)) %>% 
+    set_colnames(c("Country", "Gender", 
+                   paste0("Age ", rep(c("50-54", "55-59", "60-64")))))
+```
+
+The resulting `output` list can be easily bound into a data frame and arranged using `dplyr`. From this package we also use `spread` in order to subset by gender. Using the pipe operator `%>%` keeps our code compact and avoids unnecessary variable assignment and inefficient memory requirements.
+
+
+In the next code chunk, `vars` is assigned to be a name vector that specifies whether labor force participation is full time, part time or not prevalent. Fetching it using `grep` from the cleaned data set makes it robust against name changes in the underlying data frame and ensures consistency. Conveniently, `grep` does not necessarily require regular expressions as an input. Instead, it is able to perform matches on simple strings and, in this case, returns the column names that contain 'labor'. At this point, the meaning of the second argument in `group.share` (comment: name the first code chunk in this section) becomes clear. It is used to subset the target data frame `df` by the extracted variable names containing the string "labor". 
```

Comment:
It's kinda weird to say 'at this point the meaning of the second argument becomes clear'. It kinda sounds like we don't really know what we're doing, you know? I just think it can be phrased slightly better.

Reply:
hahaha youre right! fixed